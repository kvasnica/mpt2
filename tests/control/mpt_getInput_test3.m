function mpt_getInput_test3

% tests whether we give nice error messages when dimension of x0 is wrong:
% http://control.ee.ethz.ch/~mpt/hg/mpt-25x?cs=1d9c35868c0a

Double_Integrator
probStruct.tracking = 1;
ctrl = mpt_control(sysStruct, probStruct, 'online');

fprintf('mpt_getInput should nicely complain about wrong dimension of x0:\n');
lasterr('');
try
    u = mpt_getInput(ctrl, [1;1;1]);
    worked = 1;
catch
    worked = 0;
end
mbg_assertfalse(worked);
mbg_assertcontains(lasterr, 'Wrong size of "Options.reference". Expecting 2 elements, got 5.');
