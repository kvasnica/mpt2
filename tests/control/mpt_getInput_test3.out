Using an LTI model...
WARNING: no state constraints defined, problem can be badly scaled...

The controller will return deltaU=u(k)-u(k-1) instead of u(k) when evaluated!
mpt_getInput should nicely complain about wrong dimension of x0:
WARNING: deltaU constraints can only be satisfied if Options.Uprev is given.
WARNING: Options.reference not specified, assuming zero.
